### ISSUE TITLE
Proposal: Make structs implement Stringer interface configurable

### ISSUE THREAD

**Issue by masseelch** [facebook/ent#917]
I am using the [liip/sheriff](https://github.com/liip/sheriff) package. As stated [here](https://github.com/liip/sheriff/issues/28) and [here](https://github.com/liip/sheriff/blob/master/sheriff.go#L194) the package will skip structs that implement `Stringer`. 

The generated models of ent do implement Stringer, but i do not know why they do other than convenience. As (or if) the `String` method is not crucial for this package i propose to either remove the generated `String` method or make its generation configurable.

**Comment by a8m** [comment]
Hey @masseelch, I'm not familiar with `sheriff` and how it can be configured, but we can wrap the `Stringer` implementation with additional template and let users override it.

>  i propose to either remove the generated String method

This isn't an option, because we don't want to break users code.

**Comment by masseelch** [comment]
Wrap it into its own template to make it overidable is a valid solution for my case.

**Comment by a8m** [comment]
Great, feel free to create a PR and let me know if you need code-pointers or help with that.

**Comment by masseelch** [comment]
Well, i think we want to be able to override the template for only some models, not all of them (at least i want to). 

How to achieve that?  I think you do quite the same stuff [here](https://github.com/facebook/ent/blob/3f1545f0a9cb021992647a1e0a0bd8557c24cf82/entc/gen/template/ent.tmpl#L35)

```go
{{- /* Implement Stringer per type. */}}
	{{- $tmpl := printf "model/%s/stringer" ($.Name | snake) }}
	{{- if hasTemplate $tmpl }}
		{{- xtemplate $tmpl . }}
        {{- else }}
                // Insert the default Stringer implementation for this model
	{{- end }}
```

**Comment by masseelch** [comment]
The other possibility is to just wrap it in `{{ define "model/stringer" }}` and let the user have something like this in his own templates:

```go
{{ range $n := $.Nodes }}
    {{ if eq $n.Name "Card" }}
        {{ $receiver := $n.Receiver }}
        func ({{ $receiver }} *{{ $n.Name }}) GoString() string {
            if {{ $receiver }} == nil {
                return fmt.Sprintf("{{ $n.Name }}(nil)")
            }
            return {{ $receiver }}.String()
        }
    {{ else }}
        // Copy pase the default stringer implementation here.
    {{ end }}
{{ end }}
```

**Comment by a8m** [comment]
The `{{ define "model/stringer" }}` option is preferred - we can change/improve this in the future. Go ahead and create a PR.

**Comment by a8m** [comment]
Fixed with #921
